src/components/Calculator.js

```diff
@@ -1,166 +1,86 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import calculate from '../logic/calculate';
 
-class Calculator extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: 0,
-      next: null,
-      operation: null,
-    };
-    this.handleClick = this.handleClick.bind(this);
-  }
+export default function Calculator() {
+  const [state, setState] = useState({
+    total: 0,
```

Comment:
- [ **OPTIONAL** ] **IMPORTANT:** Good job refactoring the `Calculator` component as required for the project. But I think would be better if you use `arrow functions` rather than `normal functions` those are preferable since they are using ES6.
**I think it would be better if you also use the `function-based` on the App.js main component as it is about refactoring the entire project.** **(Arrow functions will be preferable)**